ui/app/AppLayouts/Chat/controls/community/HoldingsDropdown.qml

```diff
@@ -0,0 +1,277 @@
+import QtQuick 2.13
+import QtQuick.Layouts 1.14
+import QtQuick.Controls 2.13
+import QtGraphicalEffects 1.13
+
+import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
+import StatusQ.Components 0.1
+import StatusQ.Controls 0.1
+import StatusQ.Controls.Validators 0.1
+import StatusQ.Core.Utils 0.1 as SQ
+
+import utils 1.0
+
+StatusDropdown {
+    id: root
+
+    property real tokenAmountValue: 0
+    property string tokenName: d.defaultTokenNameText
+    property url tokenImage: ""
+    property int operator: SQ.Utils.Operators.None
+    property bool withOperatorSelector: true
+
+    signal addToken(string tokenText, url tokenImage, int operator)
+
+    function reset() {
+        root.tokenAmountValue = 0
+        root.tokenName = d.defaultTokenNameText
+        root.tokenImage = ""
+        root.operator = SQ.Utils.Operators.None
+    }
+
+    QtObject {
+        id: d
+
+        readonly property bool ready: root.tokenAmountValue > 0 && root.tokenName !== d.defaultTokenNameText
+
+        // By design values:
+        readonly property int initialHeight: 232
+        readonly property int mainHeight: 256
+        readonly property int operatorsHeight: 96
+        readonly property int extendedHeight: 417
+        readonly property int defaultWidth: 289
+        readonly property int operatorsWidth: 159
+
+        property string defaultTokenNameText: qsTr("Choose token")
```

Comment:
`readonly`?

Reply:
Mmmm, here not sure, bc the idea was having it custom for the consumer. WDYT?

Reply:
I see... good